src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantConsensusCall.java

```diff
@@ -172,41 +98,55 @@ static int getSvLength(final NovelAdjacencyReferenceLocations breakpoints, GATKS
                     }
                 }
             } else {
                 final boolean hasNoDupSeq = novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty();
                 final boolean hasNoInsertedSeq = novelAdjacencyReferenceLocations.complication.insertedSequenceForwardStrandRep.isEmpty();
                 if (hasNoDupSeq) {
                     if (hasNoInsertedSeq) {
-                        type = GATKSVVCFHeaderLines.SVTYPES.DEL; // clean deletion
+                        type = new SVTYPE.DEL(novelAdjacencyReferenceLocations); // clean deletion
                     } else {
-                        type = GATKSVVCFHeaderLines.SVTYPES.DEL; // scarred deletion
+                        type = new SVTYPE.DEL(novelAdjacencyReferenceLocations); // scarred deletion
                     }
                 } else {
                     if (hasNoInsertedSeq) {
-                        type = GATKSVVCFHeaderLines.SVTYPES.DEL; // clean contraction of repeat 2 -> 1, or complex contraction
+                        type = new SVTYPE.DEL(novelAdjacencyReferenceLocations); // clean contraction of repeat 2 -> 1, or complex contraction
                     } else {
                         throw new GATKException("Something went wrong in type inference, there's suspected deletion happening but both inserted sequence and duplication exits (not supported yet): " + novelAdjacencyReferenceLocations.toString());
                     }
                 }
             }
-            attributeMap.put(GATKSVVCFHeaderLines.SVTYPE, type.name());
-            if(!novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty())
-                attributeMap.put( (novelAdjacencyReferenceLocations.complication.dupSeqRepeatNumOnRef < novelAdjacencyReferenceLocations.complication.dupSeqRepeatNumOnCtg) ? SVConstants.CallingStepConstants.TANDUP_EXPANSION_STRING : SVConstants.CallingStepConstants.TANDUP_CONTRACTION_STRING, "");
         } else {
-            type = GATKSVVCFHeaderLines.SVTYPES.INV;
-            attributeMap.put(GATKSVVCFHeaderLines.SVTYPE, type.name());
-            attributeMap.put( (endConnectionType == FIVE_TO_FIVE) ? GATKSVVCFHeaderLines.INV55 : GATKSVVCFHeaderLines.INV33, "");
+            type = new SVTYPE.INV(novelAdjacencyReferenceLocations);
         }
-        return attributeMap;
+
+        // developer check to make sure new types are treated correctly
+        try {
+            final SVTYPE.RECOGNIZED_SVTYPES x = SVTYPE.RECOGNIZED_SVTYPES.valueOf(type.toString());
+        } catch (final IllegalArgumentException ex) {
+            throw new GATKException.ShouldNeverReachHereException("Inferred type is not known yet: " + type.toString(), ex);
+        }
+        return type;
+    }
+
+    @VisibleForTesting
+    static Map<String, String> setFlags(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations, final SVTYPE svtype) {
```

Comment:
It seems like this could also be moved to the SVTYPE subclasses -- what do you think?